shared/js/background.js

```diff
@@ -144,7 +144,11 @@ chrome.webRequest.onBeforeRequest.addListener(
         // don't have a tab instance for this tabId or this is a new requestId.
         if (requestData.type === "main_frame") {
             if (!thisTab || (thisTab.requestId !== requestData.requestId)) {
-                thisTab = tabManager.create(requestData);
+                let newTab = tabManager.create(requestData)
+
+                // persist the last URL the tab was trying to upgrade to HTTPS
+                newTab.lastHttpsUpgrade = thisTab && thisTab.lastHttpsUpgrade
```

Comment:
This was necessary in the RaisingChildren case. The redirect wasn't done with a 30* HTTP status or something, it actually served a page which redirected you to the HTTP version:

```
$ curl https://raisingchildren.net.au/articles/safe_water_temperature.html
<html><head><title></title><!-- <script language="javascript">window.location.replace("http://raisingchildren.net.au/articles/safe_water_temperature.html");</script> --></head><body></body></html>
```

This meant the `requestId` for the tab was different, which meant `thisTab` was refreshed, and so we need to persist `lastHttpsUpgrade` between the two somehow.